pyart/graph/radarmapdisplay_cartopy.py

```diff
@@ -0,0 +1,510 @@
+            Width and height of map domain in meters.
+            Only this set of parameters or the previous set of parameters
+            (min_lat, max_lat, min_lon, max_lon) should be specified.
+            If neither set is specified then the map domain will be determined
+            from the extend of the radar gate locations.
+        lon_0, lat_0 : float
+            Center of the map domain in degrees.  If the default, None is used
+            the latitude and longitude of the radar will be used.
+        shapefile : str
+            Filename for a ESRI shapefile as background (untested).
+        resolution : '10m', '50m', '110m'.
+            Resolution of NaturalEarthFeatures to use. See Cartopy
+            documentation for details.
+        gatefilter : GateFilter
+            GateFilter instance. None will result in no gatefilter mask being
+            applied to data.
+        filter_transitions : bool
+            True to remove rays where the antenna was in transition between
+            sweeps from the plot.  False will include these rays in the plot.
+            No rays are filtered when the antenna_transition attribute of the
+            underlying radar is not present.
+        edges : bool
+            True will interpolate and extrapolate the gate edges from the
+            range, azimuth and elevations in the radar, treating these
+            as specifying the center of each gate.  False treats these
+            coordinates themselved as the gate edges, resulting in a plot
+            in which the last gate in each ray and the entire last ray are not
+            not plotted.
+        embelish: bool
+            True by default. Set to false to supress drawing of coastlines
+            etc.. Use for speedup when specifying shapefiles.
+        basemap: Cartopy GeoAxes instance
+            If None, create GeoAxes instance using other keyword info.
+            If not None, use the user-specifed basemap instance.
+
+        """
+        # parse parameters
+        ax, fig = parse_ax_fig(ax, fig)
+        norm, vmin, vmax = parse_norm_vmin_vmax(
+            norm, self._radar, field, vmin, vmax)
+        cmap = parse_cmap(cmap, field)
+        if lat_lines is None:
+            lat_lines = np.arange(30, 46, 1)
+        if lon_lines is None:
+            lon_lines = np.arange(-110, -75, 1)
+        if lat_0 is None:
+            lat_0 = self.loc[0]
+        if lon_0 is None:
+            lon_0 = self.loc[1]
+
+        # get data for the plot
+        data = self._get_data(
+            field, sweep, mask_tuple, filter_transitions, gatefilter)
+        x, y = self._get_x_y(sweep, edges, filter_transitions)
+
+        # mask the data where outside the limits
+        if mask_outside:
+            data = np.ma.masked_outside(data, vmin, vmax)
+
+        proj = cartopy.crs.LambertConformal(
+            central_longitude=lon_0, central_latitude=lat_0,
+            standard_parallels=(lat_0, ))
```

Comment:
The creator or proj should be delayed until needed if no projection argument is provided.